### ISSUE TITLE
All Search Nodes are crashing

### ISSUE THREAD

**Issue by Chaitanya12** [vespa-engine/vespa#30924]
The cluster was up for couple of days, there was no new deployment done and only new data was getting ingested. The search node crashed and while trying to start again, they are failing with the same error and crashing. Below are the logs before it happened.

```
1712236533.775524       node5.hostname.com   133/37210       searchnode      proton.proton.common.eventlogger        event   state/1 name="flush.init" value="{"name":"schema_name.0.ready.memoryindex.flush"}"
1712236533.783575       node5.hostname.com   133/37210       searchnode      proton.proton.common.eventlogger        event   state/1 name="flush.start" value="{"name":"schema_name.0.ready.memoryindex.flush","memory":{"before":1098006359,"after":20250463,"tofree":1077755896},"serialnum":{"unflushed":1237112,"current":1302207}}"
1712236541.349997       node5.hostname.com   133/18139       searchnode      proton.searchcorespi.index.eventlogger  event   state/1 name="diskindex.load.start" value="{"input":{"dir":"n1/documents/schema_name/0.ready/index/index.flush.41","size":379994112}}"
1712236541.382927       node5.hostname.com   133/18139       searchnode      proton.searchcorespi.index.eventlogger  event   state/1 name="diskindex.load.complete" value="{"time.elapsed.ms":29,"input":{"dir":"n1/documents/schema_name/0.ready/index/index.flush.41","size":379994112}}"
1712236541.630080       node5.hostname.com   133/18139       searchnode      proton.proton.common.eventlogger        event   state/1 name="flush.complete" value="{"name":"schema_name.0.ready.memoryindex.flush","time.elapsed.ms":7842,"serialnum":{"flushed":1237111},"output":{"dir":"cluster.doc/n1/documents/schema_name/0.ready/index/index.flush.41","size":379994112}}"
1712236541.630351       node5.hostname.com   133/37210       searchnode      proton.proton.common.eventlogger        event   state/1 name="flush.prune" value="{"name":"schema_name.0.ready.attribute.flush.article_url","serialnum":{"oldestflushed":1087798}}"
1712236541.630900       node5.hostname.com   133/37210       searchnode      proton.proton.common.eventlogger        event   state/1 name="flush.init" value="{"name":"schema_name.0.ready.memoryindex.fusion"}"
1712236541.630961       node5.hostname.com   133/37210       searchnode      proton.proton.common.eventlogger        event   state/1 name="flush.start" value="{"name":"schema_name.0.ready.memoryindex.fusion","memory":{"before":0,"after":0,"tofree":0},"serialnum":{"unflushed":1302235,"current":1302234}}"
1712236541.644521       node5.hostname.com   133/18139       searchnode      proton.searchcorespi.index.eventlogger  event   state/1 name="fusion.start" value="{"inputs":[{"dir":"n1/documents/schema_name/0.ready/index/index.fusion.38","size":12054306816},{"dir":"n1/documents/schema_name/0.ready/index/index.flush.39","size":465805312},{"dir":"n1/documents/schema_name/0.ready/index/index.flush.40","size":419344384},{"dir":"n1/documents/schema_name/0.ready/index/index.flush.41","size":379994112}],"output":{"dir":"n1/documents/schema_name/0.ready/index/index.fusion.41","size":0}}"
1712236700.547873       node5.hostname.com   133/18139       searchnode      proton.searchcorespi.index.eventlogger  event   state/1 name="fusion.complete" value="{"time.elapsed.ms":158899,"output":{"dir":"n1/documents/schema_name/0.ready/index/index.fusion.41","size":13202350080}}"
1712236700.551592       node5.hostname.com   133/18139       searchnode      proton.searchcorespi.index.eventlogger  event   state/1 name="diskindex.load.start" value="{"input":{"dir":"n1/documents/schema_name/0.ready/index/index.fusion.41","size":13202354176}}"
1712236705.522681       node5.hostname.com   133/18139       searchnode      proton.searchcorespi.index.eventlogger  event   state/1 name="diskindex.load.complete" value="{"time.elapsed.ms":4967,"input":{"dir":"n1/documents/schema_name/0.ready/index/index.fusion.41","size":13202354176}}"
1712236705.826373       node5.hostname.com   133/18139       searchnode      proton.proton.common.eventlogger        event   state/1 name="flush.complete" value="{"name":"schema_name.0.ready.memoryindex.fusion","time.elapsed.ms":164191,"serialnum":{"flushed":1302234},"output":{"dir":"cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.41","size":13202354176}}"
1712236705.826701       node5.hostname.com   133/37210       searchnode      proton.proton.common.eventlogger        event   state/1 name="flush.prune" value="{"name":"schema_name.0.ready.attribute.flush.article_url","serialnum":{"oldestflushed":1087798}}"
1712252207.401872       node5.hostname.com   133/37210       searchnode      proton.proton.common.eventlogger        event   state/1 name="flush.init" value="{"name":"schema_name.0.ready.memoryindex.flush"}"
1712252207.408747       node5.hostname.com   133/37210       searchnode      proton.proton.common.eventlogger        event   state/1 name="flush.start" value="{"name":"schema_name.0.ready.memoryindex.flush","memory":{"before":1105634011,"after":20320535,"tofree":1085313476},"serialnum":{"unflushed":1302208,"current":1372388}}"
1712252214.929446       node5.hostname.com   133/18139       searchnode      proton.searchcorespi.index.eventlogger  event   state/1 name="diskindex.load.start" value="{"input":{"dir":"n1/documents/schema_name/0.ready/index/index.flush.42","size":391266304}}"
1712252214.967386       node5.hostname.com   133/18139       searchnode      proton.searchcorespi.index.eventlogger  event   state/1 name="diskindex.load.complete" value="{"time.elapsed.ms":34,"input":{"dir":"n1/documents/schema_name/0.ready/index/index.flush.42","size":391266304}}"
1712252215.252500       node5.hostname.com   133/18139       searchnode      proton.proton.common.eventlogger        event   state/1 name="flush.complete" value="{"name":"schema_name.0.ready.memoryindex.flush","time.elapsed.ms":7840,"serialnum":{"flushed":1302207},"output":{"dir":"cluster.doc/n1/documents/schema_name/0.ready/index/index.flush.42","size":391266304}}"
1712252215.252824       node5.hostname.com   133/37210       searchnode      proton.proton.common.eventlogger        event   state/1 name="flush.prune" value="{"name":"schema_name.0.ready.attribute.flush.article_url","serialnum":{"oldestflushed":1087798}}"
1712259491.814032       node5.hostname.com   133/28762       searchnode      proton.proton.common.eventlogger        event   state/1 name="lidspace.compaction.complete" value="{"documentsubdb":"schema_name.1.removed","lidlimit":42136}"
1712264415.380828       node5.hostname.com   137/51302       logd    logdemon        info    removing /opt/vespa/logs/vespa/vespa.log-2024-03-06.12-59-58, too old (30.000143 days)
1712269327.024298       node5.hostname.com   133/37210       searchnode      proton.proton.common.eventlogger        event   state/1 name="flush.init" value="{"name":"schema_name.0.ready.memoryindex.flush"}"
1712269327.033712       node5.hostname.com   133/37210       searchnode      proton.proton.common.eventlogger        event   state/1 name="flush.start" va
lue="{"name":"schema_name.0.ready.memoryindex.flush","memory":{"before":1097401773,"after":20404721,"tofree":1076997052},"serialnum":{"unflushed":1372389,"cu
rrent":1457197}}"
1712269334.937643       node5.hostname.com   133/18139       searchnode      proton.searchcorespi.index.eventlogger  event   state/1 name="diskindex.load.
start" value="{"input":{"dir":"n1/documents/schema_name/0.ready/index/index.flush.43","size":455303168}}"
1712269335.036015       node5.hostname.com   133/18139       searchnode      proton.searchcorespi.index.eventlogger  event   state/1 name="diskindex.load.
complete" value="{"time.elapsed.ms":94,"input":{"dir":"n1/documents/schema_name/0.ready/index/index.flush.43","size":455303168}}"
1712269335.290578       node5.hostname.com   133/18139       searchnode      proton.proton.common.eventlogger        event   state/1 name="flush.complete"
 value="{"name":"schema_name.0.ready.memoryindex.flush","time.elapsed.ms":8253,"serialnum":{"flushed":1372388},"output":{"dir":"cluster.doc/n1/documents/
schema_name/0.ready/index/index.flush.43","size":455303168}}"
1712269335.290906       node5.hostname.com   133/37210       searchnode      proton.proton.common.eventlogger        event   state/1 name="flush.prune" va
lue="{"name":"schema_name.0.ready.attribute.flush.article_url","serialnum":{"oldestflushed":1087798}}"
1712270715.427063       node5.hostname.com   133/37210       searchnode      proton.proton.common.eventlogger        event   state/1 name="flush.init" val
ue="{"name":"schema_name.0.ready.summary.compact_spread"}"
1712270715.427384       node5.hostname.com   133/37210       searchnode      proton.proton.common.eventlogger        event   state/1 name="flush.start" va
lue="{"name":"schema_name.0.ready.summary.compact_spread","memory":{"before":25439608,"after":25439608,"tofree":0},"serialnum":{"unflushed":1463915,"current"
:1463914}}"
1712270715.427457       node5.hostname.com   133/18139       searchnode      proton.searchlib.docstore.logdatastore  info    Compacting file '/opt/vespa/v
ar/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/summary/1712248607101560154' which has bloat '0.00' and bucket-spread '3.2693
1712270724.305371       node5.hostname.com   133/18139       searchnode      proton.searchlib.docstore.compacter     info    Have read 107557 lids and pla
ced them in 5639 buckets. Temporary compressed in 32079 chunks.
1712270736.559670       node5.hostname.com   133/18139       searchnode      proton.searchlib.docstore.compacter     info    Compacted 107557 lids into 56
39 buckets
1712270737.683692       node5.hostname.com   133/18139       searchnode      proton.searchlib.docstore.logdatastore  info    Done compacting due to spread
: Disk bloat is now at 2689526 of 15299493742 at 0.02 percent
1712270737.684116       node5.hostname.com   133/18139       searchnode      proton.proton.common.eventlogger        event   state/1 name="flush.complete"
 value="{"name":"schema_name.0.ready.summary.compact_spread","time.elapsed.ms":22256,"serialnum":{"flushed":1463914},"output":{"dir":"cluster.doc/n1/docu
ments/schema_name/0.ready/summary","size":15303847936}}"
1712270737.684435       node5.hostname.com   133/37210       searchnode      proton.proton.common.eventlogger        event   state/1 name="flush.prune" va
lue="{"name":"schema_name.0.ready.attribute.flush.article_url","serialnum":{"oldestflushed":1087798}}"
1712271585.912574       node5.hostname.com   133/28762       searchnode      proton.proton.server.lidspace.compactionjob     info    schema_name.1.removed
: Lid space compaction is disabled while remove batch (delete buckets) is ongoing
1712271595.939004       node5.hostname.com   133/28762       searchnode      proton.proton.server.lidspace.compactionjob     info    schema_name.1.removed: Lid space compaction is re-enabled as remove operations are no longer ongoing
1712284159.118638       node5.hostname.com   133/37210       searchnode      proton.proton.common.eventlogger        event   state/1 name="flush.init" value="{"name":"schema_name.0.ready.memoryindex.flush"}"
1712284159.129231       node5.hostname.com   133/37210       searchnode      proton.proton.common.eventlogger        event   state/1 name="flush.start" value="{"name":"schema_name.0.ready.memoryindex.flush","memory":{"before":1094426143,"after":20469899,"tofree":1073956244},"serialnum":{"unflushed":1457198,"current":1522776}}"
1712284167.881929       node5.hostname.com   133/18139       searchnode      proton.searchcorespi.index.eventlogger  event   state/1 name="diskindex.load.start" value="{"input":{"dir":"n1/documents/schema_name/0.ready/index/index.flush.44","size":609734656}}"
1712284168.112492       node5.hostname.com   133/18139       searchnode      proton.searchcorespi.index.eventlogger  event   state/1 name="diskindex.load.complete" value="{"time.elapsed.ms":227,"input":{"dir":"n1/documents/schema_name/0.ready/index/index.flush.44","size":609734656}}"
1712284168.329380       node5.hostname.com   133/18139       searchnode      proton.proton.common.eventlogger        event   state/1 name="flush.complete" value="{"name":"schema_name.0.ready.memoryindex.flush","time.elapsed.ms":9196,"serialnum":{"flushed":1457197},"output":{"dir":"cluster.doc/n1/documents/schema_name/0.ready/index/index.flush.44","size":609734656}}"
1712284168.329684       node5.hostname.com   133/37210       searchnode      proton.proton.common.eventlogger        event   state/1 name="flush.prune" value="{"name":"schema_name.0.ready.attribute.flush.article_url","serialnum":{"oldestflushed":1087798}}"
1712284168.330306       node5.hostname.com   133/37210       searchnode      proton.proton.common.eventlogger        event   state/1 name="flush.init" value="{"name":"schema_name.0.ready.memoryindex.fusion"}"
1712284168.330372       node5.hostname.com   133/37210       searchnode      proton.proton.common.eventlogger        event   state/1 name="flush.start" value="{"name":"schema_name.0.ready.memoryindex.fusion","memory":{"before":0,"after":0,"tofree":0},"serialnum":{"unflushed":1522818,"current":1522817}}"
1712284168.343925       node5.hostname.com   133/18139       searchnode      proton.searchcorespi.index.eventlogger  event   state/1 name="fusion.start" value="{"inputs":[{"dir":"n1/documents/schema_name/0.ready/index/index.fusion.41","size":13202354176},{"dir":"n1/documents/schema_name/0.ready/index/index.flush.42","size":391266304},{"dir":"n1/documents/schema_name/0.ready/index/index.flush.43","size":455303168},{"dir":"n1/documents/schema_name/0.ready/index/index.flush.44","size":609734656}],"output":{"dir":"n1/documents/schema_name/0.ready/index/index.fusion.44","size":0}}"
1712284334.107713       node5.hostname.com   133/18139       searchnode      proton.searchcorespi.index.eventlogger  event   state/1 name="fusion.complete" value="{"time.elapsed.ms":165760,"output":{"dir":"n1/documents/schema_name/0.ready/index/index.fusion.44","size":14521040896}}"
1712284334.111694       node5.hostname.com   133/18139       searchnode      proton.searchcorespi.index.eventlogger  event   state/1 name="diskindex.load.start" value="{"input":{"dir":"n1/documents/schema_name/0.ready/index/index.fusion.44","size":14521044992}}"
1712284335.079171       node5.hostname.com   133     searchnode      stderr  warning vespa-proton-bin: /builddir/build/BUILD/vespa-8.270.8/searchlib/src/vespa/searchlib/diskindex/pagedict4randread.cpp:237: virtual bool search::diskindex::PageDict4RandRead::open(const vespalib::string&, const search::TuneFileRandRead&): Assertion `_ssReadContext.getBufferEndFilePos() >= fileSize' failed.
1712284346.137249       node5.hostname.com   107/41690       config-sentinel sentinel.sentinel.service       event   crash/1 name="searchnode" pid=133 signal="Aborted"
1712284346.137405       node5.hostname.com   107/41690       config-sentinel sentinel.sentinel.service       info    searchnode: incremented restart penalty to 2.000 seconds
1712284346.139986       node5.hostname.com   457/41690       config-sentinel sentinel.sentinel.service       event   starting/1 name="searchnode"
```

**Comment by toregge** [comment]
This is likely caused by many long words in the disk index, causing the `.ssdat` file for a disk index dictionary to become larger than 4 GiB. PR #30926 contains relevant fixes.

**Comment by Chaitanya12** [comment]
The output of `find /opt/vespa/var/db/vespa/search/cluster.neondoc/n1/documents/owl_matters/ -name '*.ssdat' -ls` can be found below:

```
134439210      4 -rw-r--r--   1  900000999 900000999     4096 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/title/dictionary.ssdat
280317118      4 -rw-r--r--   1  900000999 900000999     4096 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/data/dictionary.ssdat
402658077      4 -rw-r--r--   1  900000999 900000999     4096 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/page_summary/dictionary.ssdat
    43510      4 -rw-r--r--   1  900000999 900000999     4096 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/mediaType/dictionary.ssdat
134442022      8 -rw-r--r--   1  900000999 900000999     4232 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/id/dictionary.ssdat
277693890      8 -rw-r--r--   1  900000999 900000999     4464 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/accounts/dictionary.ssdat
402921130 4293320 -rw-r--r--   1  900000999 900000999 4396257680 Apr  4 22:32 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/activities/dictionary.ssdat
    43516       8 -rw-r--r--   1  900000999 900000999       4112 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/deals/dictionary.ssdat
134439216       8 -rw-r--r--   1  900000999 900000999       4120 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/distributionLists/dictionary.ssdat
281327498       8 -rw-r--r--   1  900000999 900000999       4400 Apr  4 22:30 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/focalActors/dictionary.ssdat
402723545       8 -rw-r--r--   1  900000999 900000999       4272 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/issuers/dictionary.ssdat
    44099       8 -rw-r--r--   1  900000999 900000999       4168 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/matterComments/dictionary.ssdat
134439222       4 -rw-r--r--   1  900000999 900000999       4096 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/matterId/dictionary.ssdat
280338265       8 -rw-r--r--   1  900000999 900000999       4120 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/matterIssues/dictionary.ssdat
402658083       8 -rw-r--r--   1  900000999 900000999       4120 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/matterReferrals/dictionary.ssdat
 42617       8 -rw-r--r--   1  900000999 900000999       4936 Apr  4 22:30 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/nonFocalActors/dictionary.ssdat
134439156       8 -rw-r--r--   1  900000999 900000999       7792 Apr  4 22:30 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/observation/dictionary.ssdat
280338277       8 -rw-r--r--   1  900000999 900000999       4200 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/participants/dictionary.ssdat
402658089       8 -rw-r--r--   1  900000999 900000999       4120 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/queues/dictionary.ssdat
    44105       8 -rw-r--r--   1  900000999 900000999       4184 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/restrictions/dictionary.ssdat
134439104       8 -rw-r--r--   1  900000999 900000999       4488 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/securities/dictionary.ssdat
280338271       8 -rw-r--r--   1  900000999 900000999       4112 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/securityOfferings/dictionary.ssdat
402658095       8 -rw-r--r--   1  900000999 900000999       4120 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/accessClassification/dictionary.ssdat
    42577       8 -rw-r--r--   1  900000999 900000999       4944 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/activityReferences/dictionary.ssdat
134439228       8 -rw-r--r--   1  900000999 900000999       4208 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/actorReferences/dictionary.ssdat
281327504       8 -rw-r--r--   1  900000999 900000999       4336 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/alertingSecurities/dictionary.ssdat
402658101       4 -rw-r--r--   1  900000999 900000999       4096 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/qaAssignedToUser/dictionary.ssdat
    44111       8 -rw-r--r--   1  900000999 900000999       4112 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/caseAssignedToUser/dictionary.ssdat
134441986       8 -rw-r--r--   1  900000999 900000999       4128 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/childMatterId/dictionary.ssdat
280338283       8 -rw-r--r--   1  900000999 900000999       4136 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/closingComment/dictionary.ssdat
402658107       8 -rw-r--r--   1  900000999 900000999       4112 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/closingResolution/dictionary.ssdat
    44118       8 -rw-r--r--   1  900000999 900000999       4112 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/country/dictionary.ssdat
134441992       8 -rw-r--r--   1  900000999 900000999       4112 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/dispositionNarrative/dictionary.ssdat
280338289       8 -rw-r--r--   1  900000999 900000999       4112 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/filingNarrative/dictionary.ssdat
402723521       8 -rw-r--r--   1  900000999 900000999       4112 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/decisionType/dictionary.ssdat
44124       8 -rw-r--r--   1  900000999 900000999       4120 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/group/dictionary.ssdat
134441998       8 -rw-r--r--   1  900000999 900000999       4128 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/parentMatterId/dictionary.ssdat
280338296       8 -rw-r--r--   1  900000999 900000999       4120 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/region/dictionary.ssdat
402723527       8 -rw-r--r--   1  900000999 900000999       4112 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/regulatoryFilingType/dictionary.ssdat
    30848       8 -rw-r--r--   1  900000999 900000999       4120 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/qaStatus/dictionary.ssdat
134442004       8 -rw-r--r--   1  900000999 900000999       4112 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/sarFilingInformation/dictionary.ssdat
280338302       8 -rw-r--r--   1  900000999 900000999       4112 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/sarJointFilingInformation/dictionary.ssdat
402723533       8 -rw-r--r--   1  900000999 900000999       4112 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/submitter/dictionary.ssdat
    30854       8 -rw-r--r--   1  900000999 900000999       4112 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/qaClosingComment/dictionary.ssdat
134442010       8 -rw-r--r--   1  900000999 900000999       4112 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/qaClosingResolution/dictionary.ssdat
281327492       8 -rw-r--r--   1  900000999 900000999       4120 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/observationFeedback/dictionary.ssdat
402723539       4 -rw-r--r--   1  900000999 900000999       4096 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/detectionNarrative/dictionary.ssdat
    30860       8 -rw-r--r--   1  900000999 900000999       4120 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/regulatoryInquiry/dictionary.ssdat
134442016       8 -rw-r--r--   1  900000999 900000999       4112 Apr  4 22:29 /opt/vespa/var/db/vespa/search/cluster.doc/n1/documents/schema_name/0.ready/index/index.fusion.44/logDecision/dictionary.ssdat
```

**Comment by toregge** [comment]
vespa 8.331.34 contains the fixes from PR #30926 and should not crash with large disk index dictionary `.ssdat` files.

**Comment by geirst** [comment]
The fix for this issue is part of Vespa 8.331.34 (available open source https://hub.docker.com/r/vespaengine/vespa/tags).

The disk index has a dictionary of unique tokens per field. Tokens that are longer than 4K are stored in a _.ssdat_ file. For the _activities_ field, this _.ssdat_ file became larger than 4 GiB. The code reading the dictionary files did not handle _.ssdat_ files larger than 4 GiB, causing the crash. This is now fixed.

However, having so many long tokens in a field is an anomaly, and should be looked into by the application owner. We have also created https://github.com/vespa-engine/vespa/issues/30942 to stop long tokens to enter the index in the first place.
